### PR TITLE
Add pull request templates

### PR DESCRIPTION
**Pull request by MarvinLira**
### Issue

We currently already have templates for issues but we have no templates for PRs yet.

### Description of the Change

I have added a template for PRs based on the templates in the  [repository of Atom](https://github.com/atom/.github/tree/master/.github/PULL_REQUEST_TEMPLATE).

### Alternate Designs

I started by creating specific templates for PRs as Atom also uses multiple templates.
But those templates didn't differ much and I decided to merge them into a single template.

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE/generic.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<!--
+  Have you read our Code of Conduct?
+  By filing an issue or pull request, you are expected to comply with it, including treating everyone with respect:
+  https://github.com/FreeProving/guidelines/blob/master/CODE_OF_CONDUCT.md
```
Comment (just95) [inline PULL_REQUEST_TEMPLATE/generic.md]: The default branch has recently been renamed from `master` to `main` (see #2).

```suggestion
  https://github.com/FreeProving/guidelines/blob/main/CODE_OF_CONDUCT.md
```

File: PULL_REQUEST_TEMPLATE/generic.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+<!--
+
+Give a short summary of the changes you made to handle the issue.
+Keep in mind that the maintainer reviewing this PR may not be familiar with or have worked with the code here recently, so please walk us through the concepts.
+
+-->
+
+### Alternate Designs
+
+<!-- Explain what other alternates were considered and why the proposed version was selected -->
```
Comment (just95) [inline PULL_REQUEST_TEMPLATE/generic.md]:
```suggestion
<!-- Explain what other alternates were considered and why the proposed version was selected. -->
```

File: PULL_REQUEST_TEMPLATE/generic.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+<!--
+
+Link to the issue that this pull request addresses.
+
+If there is not yet a corresponding issue, please open a new issue and then link to that issue in your pull request.
+
+Give any additional information that is not covered by the linked issue but might be important for the reviewer.
+
+-->
```
Comment (just95) [inline PULL_REQUEST_TEMPLATE/generic.md]: The GitHub issue and pull request comment editor does not provide any syntax highlighting. Thus, it is difficult to tell where Markdown comments start and where they end. Therefore, I would avoid blank lines in Markdown comments such that the user does not insert the text between `<!-- … -->` by accident. Furthermore, I used some indentation in the issue templates to make start and end of the comment stand out more.

```suggestion
<!--
  Link to the issue that this pull request addresses.
  If there is not yet a corresponding issue, please open a new issue and then link to that issue in your pull request.
  Give any additional information that is not covered by the linked issue but might be important for the reviewer.
-->
```

**Review by just95 — CHANGES_REQUESTED** [review]

**Comment by just95** [comment]
Oh, and we should maybe also mention the template in the *Create a Pull Request* section of the [CONTRIBUTING.md](https://github.com/FreeProving/guidelines/blob/main/CONTRIBUTING.md#contributing-code).

**Review by just95 — APPROVED** [review]

**Review by ichistmeinname — APPROVED** [review]

**Pull request merged**